src/move_basic.cpp

```diff
@@ -731,22 +755,36 @@ bool MoveBasic::moveLinear(const tf2::Transform& goalInDriving,
         ros::spinOnce();
         r.sleep();
 
-        if (!getTransform(drivingFrame, baseFrame, poseDriving)) {
+        // Try to update the goal in the driving frame. This might not work,
+        // for example if it was based on a fiducial which was no longer
+        // visible. However, if the goal was based up a fiducial, then it
+        // is likley that the estimate of our position relative to it
+        // will improve as we get closer
+        tf2::Transform T_planning_driving;
+        if (getTransform(planningFrame, drivingFrame, T_planning_driving)) {
+            goalInDriving = T_planning_driving * goalInPlanning;
+            if (verboseInfo) {
+                double gx, gy, gyaw;
+                getPose(goalInDriving, gx, gy, gyaw);
+                printf("Updated goal %f %f %f\n", gx, gy, gyaw);
+            }
+        }
+        else if (verboseInfo) {
+            printf("Could not update goal\n");
+        }
+
+        if (!getTransform("odom", baseFrame, poseDriving)) {
              ROS_WARN("MoveBasic: Cannot determine robot pose for linear");
              continue;
         }
-
         goalInBase = poseDriving * goalInDriving;
         remaining = goalInBase.getOrigin();
-        if (!forward) {
-           remaining = -remaining;
-        }
+        double distRemaining = sqrt(linear.x() * linear.x() + linear.y() * linear.y());
```

Comment:
`std::hypot` might make sense here.

Reply:
I didn't know about `std::hypot`, good to know.  I changed it to zero `z` and use `Vector3::length()` again.